python/hail/methods/impex.py

```diff
@@ -764,13 +764,179 @@ def import_gen(path, sample_file=None, tolerance=0.2, min_partitions=None, chrom
+
+    Importing a table with explicit type annotations:
+
+    >>> table = hc1.import_table('data/samples1.tsv', types={'Height': TFloat64(), 'Age': TInt32()})
+
+    Note that string columns like ``Sample`` and ``Status`` do not need to be
+    typed, because ``String`` is the default type.
+
+    Imputing a table with type imputation (the file must be read twice to
+    impute):
+
+    >>> table = hc1.import_table('data/samples1.tsv', impute=True)
+
+    **Detailed examples**
+
+    Let's import annotations from a CSV file with missing data and special characters:
+
+    .. code-block:: text
+
+        $ cat data/samples2.tsv
+        Batch,PT-ID
+        1kg,PT-0001
+        1kg,PT-0002
+        study1,PT-0003
+        study3,PT-0003
+        .,PT-0004
+        1kg,PT-0005
+        .,PT-0006
+        1kg,PT-0007
+
+    In this case, we should:
+
+    - Pass the non-default delimiter ``,``
+
+    - Pass the non-default missing value ``.``
+
+    >>> table = hc1.import_table('data/samples2.tsv', delimiter=',', missing='.')
+
+    Let's import annotations from a file with no header and sample IDs that need
+    to be transformed.  Suppose the vds sample IDs are of the form
+    ``NA#####``. This file has no header line, and the sample ID is hidden in a
+    field with other information.
+
+    .. code-block: text
+
+        $ cat data/samples3.tsv
+        1kg_NA12345   female
+        1kg_NA12346   male
+        1kg_NA12348   female
+        pgc_NA23415   male
+        pgc_NA23418   male
+
+    To import:
+
+    >>> annotations = (hc1.import_table('data/samples3.tsv', no_header=True)
+    ...                   .annotate('sample = f0.split("_")[1]')
+    ...                   .key_by('sample'))
+
+    **Notes**
+
+    The `impute` parameter tells Hail to scan the file an extra time to gather
+    information about possible field types. While this is a bit slower for large
+    files, (the file is parsed twice), the convenience is often worth this cost.
+
+    The `delimiter` parameter is either a delimiter character (if a single
+    character) or a field separator regex (2 or more characters). This regex
+    follows the `Java regex standard
+    <http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html>`_.
+
+    .. note::
+
+        Use ``delimiter='\\s+'`` to specify whitespace delimited files.
+
+    `comment` optionally causes Hail to skip any line that starts with the given
+    string. Passing ``comment='#'`` will skip any line beginning in a pound
+    sign, for example.
+
+    `missing` defines the representation of missing data in the table.
+
+    .. note::
+
+        The `comment` and `missing` parameters are **NOT** regexes.
+
+    The `no_header` parameter indicates that the file has no header line. If
+    this option is passed, then the column names will be `f0`, `f1`,
+    ... `fN` (0-indexed).
+
+    The `types` parameter allows the user to pass the types of columns in the
```

Comment:
``types``

Reply:
?

Reply:
nm, I think you're right about single backticks